Sources/MongoSwift/BSON/BSONEncoder.swift

```diff
@@ -329,6 +360,29 @@ extension _BSONEncoder {
         return try self.box_(value) ?? Document()
     }
 
+    fileprivate func handleCustomStrategy<T: Encodable>(
```

Comment:
good idea pulling this out into a separate method! it seems like Apple just copy-pasted this block instead.... lol.

my only comment is that I wonder if we can make the parameter labels a little clearer so it reads better at the call site. I see why you named the first parameter `encode` since it's basically an alternate `encode(to:)` method, but from where we use this I'm not sure that's obvious and it kinda reads more like we're encoding `f`. at the least I think we should change that label to something clearer like `encodeFunc`, or maybe do something like `handleCustomStrategy(f, forValue: data)`? or `handleCustom(strategy: f, forValue: data)`? up to you.

Reply:
I like `encodeFunc` and `forValue`; they seem much clearer than what I have. Changing it to them 👍